chanfitness/revenue.go

```diff
@@ -1,10 +1,77 @@
 package chanfitness
 
 import (
+	"errors"
+
 	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/channeldb"
 	"github.com/lightningnetwork/lnd/lnwire"
 )
 
+var (
+	// maxQueryEvents is the number of events to query the forward log for at
+	// a time.
+	maxQueryEvents uint32 = 500
+
+	// errUnknownChannelID is returned when we cannot map a short channel ID to
+	// a channel outpoint.
+	errUnknownChannelID = errors.New("cannot find channel outpoint")
+)
+
+// eventsQuery is a function which returns paginated queries for forwarding
+// events.
+type eventsQuery func(
```

Comment:
alternative method signature could be 

```go
query func(func(event ForwardingEvent) error) error
```
such that each event would be provided in the callback. Offset tracking would still need to be done on a higher level though.